src/components/divider/divider.ts

```diff
@@ -0,0 +1,79 @@
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { themes } from '../../theming/theming-decorator.js';
+import { registerComponent } from '../common/definitions/register.js';
+import { styles } from './themes/divider.base.css.js';
+import { styles as shared } from './themes/shared/divider.common.css.js';
+import { all } from './themes/themes.js';
+
+/**
+ * The igc-divider allows the content author to easily create a horizontal/vertical rule as a break between content to better organize information on a page.
+ *
+ * @element igc-divider
+ *
+ */
+
+@themes(all)
+export default class IgcDividerComponent extends LitElement {
```

Comment:
```suggestion
/**
 * The igc-divider allows the content author to easily create a horizontal/vertical rule as a break between content to better organize information on a page.
 *
 * @element igc-divider
 *
 * @cssproperty --color - Sets the color of the divider.
 * @cssproperty --inset - Shrinks the divider by the given amount from the start. If `middle` is set it will shrink from both sides.
 */
@themes(all)
export default class IgcDividerComponent extends LitElement {
```

Please, add the the CSS variables to the class API docs.